proxy/src/coco/peer.rs

```diff
@@ -0,0 +1,617 @@
+    let default_branch = project.default_branch();
+    let repo = peer.storage().open_repo(project.urn())?;
+    let mut browser = repo.browser(default_branch)?;
+    callback(&mut browser)
+}
+
+/// Initialize a [`project::Project`] that is owned by the `owner`.
+/// This kicks off the history of the project, tracked by `librad`'s mono-repo.
+///
+/// # Errors
+///
+/// Will error if:
+///     * The signing of the project metadata fails.
+///     * The interaction with `librad` [`librad::git::storage::Storage`] fails.
+pub fn init_project(
+    peer: &PeerApi,
+    owner: &User,
+    path: impl AsRef<std::path::Path> + Send,
+    name: &str,
+    description: &str,
+    default_branch: &str,
+) -> Result<project::Project<entity::Draft>, error::Error> {
+    // Test if the repo has setup rad remote.
+    if let Ok(repo) = git2::Repository::open(&path) {
+        if repo.find_remote("rad").is_ok() {
+            return Err(error::Error::RadRemoteExists(format!(
+                "{}",
+                path.as_ref().display(),
+            )));
+        }
+    }
+
+    let meta: Result<project::Project<entity::Draft>, error::Error> = {
+        let key = peer.key();
+
+        // Create the project meta
+        let mut meta = project::Project::<entity::Draft>::create(name.to_string(), owner.urn())?
+            .to_builder()
+            .set_description(description.to_string())
+            .set_default_branch(default_branch.to_string())
+            .add_key(key.public())
+            .build()?;
+        meta.sign_owned(key)?;
+        let urn = meta.urn();
+
+        let storage = peer.storage().reopen()?;
+
+        if storage.has_urn(&urn)? {
+            return Err(error::Error::EntityExists(urn));
+        } else {
+            let _repo = storage.create_repo(&meta)?;
+        }
+        Ok(meta)
+    };
+
+    // Doing ? above breaks inference. Gaaaawwwwwd Rust!
+    let meta = meta?;
+
+    setup_remote(peer, path, &meta.urn().id, default_branch)?;
+
+    Ok(meta)
+}
+
+/// Create a [`user::User`] with the provided `handle`. This assumes that you are creating a
+/// user that uses the secret key the `PeerApi` was configured with.
+///
+/// # Errors
+///
+/// Will error if:
+///     * The signing of the user metadata fails.
+///     * The interaction with `librad` [`librad::git::storage::Storage`] fails.
+pub fn init_user(peer: &PeerApi, handle: &str) -> Result<user::User<entity::Draft>, error::Error> {
+    let key = peer.key();
+
+    // Create the project meta
+    let mut user = user::User::<entity::Draft>::create(handle.to_string(), key.public())?;
+    user.sign_owned(key)?;
+    let urn = user.urn();
+
+    let storage = peer.storage().reopen()?;
+
+    if storage.has_urn(&urn)? {
+        return Err(error::Error::EntityExists(urn));
+    } else {
+        let _repo = storage.create_repo(&user)?;
+    }
+
+    Ok(user)
+}
+
+/// Verify a user using a fake resolver that resolves the user to itself.
```

Comment:
Should we leave a TODO and/or a follow up issue for this?

Reply:
The follow-up issue is to integrate real resolvers. I can leave a TODO :ok_hand: